src/goto-symex/goto_symex_state.cpp

```diff
@@ -690,18 +690,15 @@ void goto_symex_statet::rename(
     }
   }
 
-  if(type.id()==ID_array)
+  if(const auto &array_type = type_try_dynamic_cast<array_typet>(type))
```

Comment:
I guess this fits more with the current way of doing things, so, yes.

Reply:
It provides extra validation beyond just checking the id, which depends on the type you are casting to

Reply:
I am in favour of using `type_try_dynamic_cast`, because it means that you only need to know the name of the type you want to cast to, you don't need to know that the `.id()` is supposed to be `ID_array` for an `array_typet`, in order to do the cast. It also reduces the likely-hood of mismatches between the check and the cast. There is little chance of checking for `ID_array` and casting to `bool_typet`, when both operations are done by the `try_cast`. I would however be in favour of renaming `type_try_dynamic_cast` to something less verbose.